core/services/workflows/syncer/orm.go

```diff
@@ -0,0 +1,96 @@
+package syncer
+
+import (
+	"context"
+	"encoding/base64"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/sqlutil"
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+)
+
+type ORM interface {
+	// GetSecretsURLByID returns the secrets URL for the given ID.
+	GetSecretsURLByID(ctx context.Context, id int64) (string, error)
+
+	// GetContents returns the contents of the secret at the given plain URL.
+	GetContents(ctx context.Context, url string) (string, error)
+
+	// Update updates the contents of the secret at the given plain URL or inserts a new record if not found.
+	Update(ctx context.Context, secretsURL, contents string) (int64, error)
+
+	// SecretsFor returns a map of secrets for the given workflowOwner and workflowName.
+	SecretsFor(ctx context.Context, workflowOwner, workflowName string) (map[string]string, error)
+}
+
+type WorkflowRegistryDS = ORM
+
+type orm struct {
+	ds   sqlutil.DataSource
+	lggr logger.Logger
+}
+
+var _ ORM = (*orm)(nil)
+
+func NewWorkflowRegistryDS(ds sqlutil.DataSource, lggr logger.Logger) *orm {
+	return &orm{
+		ds:   ds,
+		lggr: lggr,
+	}
+}
+
+func (orm *orm) GetSecretsURLByID(ctx context.Context, id int64) (string, error) {
+	var secretsURL string
+	if err := orm.ds.GetContext(ctx, &secretsURL,
+		`SELECT secrets_url FROM workflow_secrets WHERE workflow_secrets.id = $1`,
```

Comment:
Do we need to base64 encode the secrets_url? Why not store it in plaintext in the DB?